### PR TITLE
Feature/dm 6883 update importer layout to be consistent with builder layout in onboarders

### PR DESCRIPTION
**Pull request by kylejmorin**
This makes the onboarders in DM be consistent visually to the nfg_onboarder builder layout updates made recently. 

Specifically, it puts the builder-layout classes and structure in place without changing functionality. I brought over the classes and structure while leaving the functionality and links in the importer as they were.

This corresponds on DM: https://github.com/network-for-good/donor_management/pull/1923

### CODE REVIEW

File: app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml [hunk 1]
```diff
@@ -4,18 +4,27 @@
 - back_button_text = local_assigns[:back_button_text]
 - submit_button_text = local_assigns[:submit_button_text]
 
-.p-3.bg-white.border-bottom
+.builder-nav
   .row.align-items-center.justify-content-between
-    .col-12.col-lg-8.py-1.mx-auto.order-lg-2
+    .col-6.col-lg.d-none.d-lg-block
+      = ui.nfg :button, :lg, :secondary, :outlined, href: previous_wizard_path, body: back_button_text.html_safe, left_icon: 'caret-left', class: 'flex-fill', disable_with: ui.nfg(:icon, :loader), render_unless: (first_step or controller.single_use_steps.include?(controller.previous_step) or at_point_of_no_return?)
```
Comment (jonathanroehm) [inline app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml:10]: @kylejmorin should flex-fill be a trait for buttons?

Comment (kylejmorin) [reply app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml:10]: @jonathanroehm As of now, i don't think so. It's used rarely right now.

File: app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml [hunk 2]
```diff
@@ -4,18 +4,27 @@
...
+      = ui.nfg :button, :lg, *traits, body: submit_button_text.html_safe, icon: 'caret-right', class: 'flex-fill', confirm: next_step_confirmation, disable_with: ui.nfg(:icon, :loader)
+
+.builder-nav-sm
+  .row
+    .col
+      = ui.nfg :button, :lg, :block, :secondary, :outlined, href: previous_wizard_path, body: back_button_text.html_safe, left_icon: 'caret-left', class: 'flex-fill', disable_with: ui.nfg(:icon, :loader), render_unless: (first_step or controller.single_use_steps.include?(controller.previous_step) or at_point_of_no_return?)
+    .col
+      - traits = [:submit]
+      - traits << :disabled if step.to_sym == :field_mapping && !import.ready_to_import?
+      = ui.nfg :button, :lg, :block, *traits, body: submit_button_text.html_safe, icon: 'caret-right', class: 'flex-fill', confirm: next_step_confirmation, disable_with: ui.nfg(:icon, :loader)
```
Comment (jonathanroehm) [inline app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml]: @kylejmorin  i went ahead and moved this traits logic to the navbar presenter and wrote a spec for it.

Comment (kylejmorin) [reply app/views/nfg_csv_importer/layouts/onboarding/import_data/_navigation_bar.html.haml]: Thank you for consolidating it! #DRY

File: app/presenters/nfg_csv_importer/onboarder/navigation_bar_presenter.rb [hunk 3]
```diff
@@ -34,6 +34,13 @@ def show_nav?
         active_step.to_sym != first_step
       end
 
+      def next_button_traits(step:, import:)
+        [
+          :submit,
+          (:disabled if step.to_sym == :field_mapping && !import.ready_to_import?)
+        ].reject(&:nil?)
```
Comment (pathakh) [inline app/presenters/nfg_csv_importer/onboarder/navigation_bar_presenter.rb:41]: if you want to remove any nil values compact method on array removes them.

Comment (kylejmorin) [reply app/presenters/nfg_csv_importer/onboarder/navigation_bar_presenter.rb:41]: @jonathanroehm do you want to add anything? This was your change.

Comment (jonathanroehm) [reply app/presenters/nfg_csv_importer/onboarder/navigation_bar_presenter.rb]: thanks @pathakh i forgot about compact. I've updated

**Pull request merged**